azurecontainerapps/azurecontainerapps.ts

```diff
@@ -78,14 +78,39 @@ export class azurecontainerapps {
                 await new ContainerRegistryHelper().loginAcrWithAccessTokenAsync(acrName);
             }
 
+            // Signals whether the Oryx builder should be used to create a runnable application image
+            let shouldUseBuilder: boolean = false;
+
+            // Signals whether an image will be created locally and pushed to ACR to use for the Container App
+            let shouldBuildAndPushImage = !util.isNullOrEmpty(appSourcePath);
+
             // Get Dockerfile to build, if provided, or check if one exists at the root of the provided application
             let dockerfilePath: string = tl.getInput('dockerfilePath', false);
+            if (shouldBuildAndPushImage) {
+                if (util.isNullOrEmpty(dockerfilePath)) {
+                    console.log(tl.loc('CheckForAppSourceDockerfileMessage', appSourcePath));
+                    const rootDockerfilePath = path.join(appSourcePath, 'Dockerfile');
+                    if (fs.existsSync(rootDockerfilePath)) {
```

Comment:
Not a big deal if it's going to be a can of worm, but is there a specific reason to do this `existsSync` operation synchronously?

Reply:
No specific reason -- I believe I found another task in the monorepo that was doing this and decided to follow the same pattern. I'm going to leave it as-is for now if that's OK.

Reply:
OK seems fair